src/user.py

```diff
@@ -0,0 +1,176 @@
+            'error': "account with this email already exists"
+        }), HTTP_409_CONFLIT
+        
+    password_hash = generate_password_hash(password)
+    
+    new_user = UserProfile(
+        id=uuid.uuid4(),
+        first_name=first_name,
+        last_name=last_name,
+        email=email,
+    )
+    # user Login
+    user_login = UserLogin(
+        id=uuid.uuid4(),
+        google_login=True,
+        is_active=True,
+        password_hash=password_hash,
+        user_profile_id=new_user.id,
+        last_login=datetime.now()
+    )
+
+    db.session.add(new_user)
+    db.session.add(user_login)
+    db.session.commit()
+    
+    # TODO: Create access and refresh tokens.
+    
+    user_info = get_user_info(new_user.id)
+    
+    return jsonify(user_info), HTTP_201_CREATED
+
+
+
+@user_bp.post('/login/user')
+@swag_from('./docs/login/user.yml')
+def login():
+    """Login endpoint
+
+    Returns:
+        json: user's information
+    """
+    
+    if 'email' not in request.json:
+        return jsonify({
+            'error': "email cannot be blank"
+        }), HTTP_400_BAD_REQUEST
+        
+    if 'password' not in request.json:
+        return jsonify({
+            'error': "password is missing"
+        }), HTTP_400_BAD_REQUEST
+    
+    email = request.json['email']
+    password = request.json['password']
+    
+    if not validators.email(email):
+        return jsonify({
+            'error': "email is invalid"
+        }), HTTP_400_BAD_REQUEST
+
+    # check if user exists
+    usr = UserProfile.query.filter_by(email=email).first()
+    if not usr:
+        return jsonify({
+            'error': "user is invalid"
+        }), HTTP_401_UNAUTHORIZED
+    
+    user_login = UserLogin.query.filter_by(user_profile_id=usr.id).first()
+    
+    
+    # Check if password is correct
+    is_valid_pass = check_password_hash(
+        pwhash=user_login.password_hash,
+        password=password)
+    if is_valid_pass:
+        # TODO: Create access and refresh tokens.
+        user_info =  get_user_info(usr.id)
+        
+        # updates last login
+        user_login.last_login = datetime.now()   
+        user_login.google_login =  False
+        db.session.commit()
+        
+        return jsonify(user_info), HTTP_200_OK
+    
+    return jsonify({'error':'incorrect password'}), HTTP_401_UNAUTHORIZED
+
+
+def validate_password_complexity(password):
+    """Checks password complexity
+    
+    Passwords:
+        Should have at least one number.
+        Should have at least one uppercase and one lowercase character.
+        Should have at least one special symbol.
+        Should be at least 8 characters long.
+        
+    Args:
+        password (string): [description]
+
+    Returns:
+        boolean: true if the conditions are met, false otherwise.
+    """
+    pattern = re.compile("^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&.,()]{8,}$")
+    match = re.search(pattern, password)
```

Comment:
Does this return a boolean value? If yes, just do  `return re.search(pattern, password)`